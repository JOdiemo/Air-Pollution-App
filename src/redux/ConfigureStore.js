import {
  createStore, combineReducers, applyMiddleware, compose,
} from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import countryReducer from './Reducers/CountrySlice';
import pollutionReducer from './Reducers/PollutionSlice';

const rootReducer = combineReducers({
  countryReducer,
  pollutionReducer,
});

const store = createStore(rootReducer,
  compose(applyMiddleware(thunk, logger)));
export default store;
