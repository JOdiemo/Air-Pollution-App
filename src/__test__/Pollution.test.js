import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import pollutionReducer from '../redux/Reducers/PollutionSlice';
import Pollutions from '../components/Pollutions';

const rootReducer = combineReducers({
  pollutionReducer,
});
const store = createStore(rootReducer, applyMiddleware(thunk));
describe('Pollutions List ', () => {
  test('Pollution lists render correctly', () => {
    const pollutions = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Pollutions />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(pollutions).toMatchSnapshot();
  });
});
