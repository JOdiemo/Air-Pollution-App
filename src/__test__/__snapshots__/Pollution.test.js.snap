// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pollutions List  Pollution lists render correctly 1`] = `
<div>
  <div>
    <nav
      className="bg-pink"
    >
      <div
        className="d-flex justify-content-between align-items-center bg- p-3 mb-2 bg-pink"
      >
        <button
          className="btn mt-2"
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/countries"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle-arrow-left icon"
              data-icon="circle-arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM384 288H205.3l49.38 49.38c12.5 12.5 12.5 32.75 0 45.25s-32.75 12.5-45.25 0L105.4 278.6C97.4 270.7 96 260.9 96 256c0-4.883 1.391-14.66 9.398-22.65l103.1-103.1c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L205.3 224H384c17.69 0 32 14.33 32 32S401.7 288 384 288z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </button>
        <h1
          className="header"
        >
          Global Air Pollution Updates
        </h1>
        <div
          className="d-flex gap-3"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-microphone link text-light"
              data-icon="microphone"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M192 352c53.03 0 96-42.97 96-96v-160c0-53.03-42.97-96-96-96s-96 42.97-96 96v160C96 309 138.1 352 192 352zM344 192C330.7 192 320 202.7 320 215.1V256c0 73.33-61.97 132.4-136.3 127.7c-66.08-4.169-119.7-66.59-119.7-132.8L64 215.1C64 202.7 53.25 192 40 192S16 202.7 16 215.1v32.15c0 89.66 63.97 169.6 152 181.7V464H128c-18.19 0-32.84 15.18-31.96 33.57C96.43 505.8 103.8 512 112 512h160c8.222 0 15.57-6.216 15.96-14.43C288.8 479.2 274.2 464 256 464h-40v-33.77C301.7 418.5 368 344.9 368 256V215.1C368 202.7 357.3 192 344 192z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            aria-current="page"
            className="link text-light active"
            href="/"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-gear icon"
              data-icon="gear"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              text="reload"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="col-lg-3 col-md-4 col-6 col-6 border border-dark"
  >
    <div
      className="d-flex justify-content-between"
    >
      <div
        className="d-flex flex-column"
      >
        <img
          alt="flag"
          className="m-2 img"
        />
        <div
          className="country-details"
        >
          <p
            className="name"
          />
          <p>
            Population:
             
          </p>
        </div>
      </div>
      <div>
        <button
          className="btn mt-2 float-right"
          onClick={[Function]}
          type="button"
        >
          <a
            className="link m-2 text-light"
            href="/pollution"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle-arrow-right icon"
              data-icon="circle-arrow-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
